pkg/types/types.go

```diff
@@ -18,11 +18,12 @@ type NetConf struct {
 	VFID          int
 	HostIFNames   string // VF netdevice name(s)
 	ContIFNames   string // VF names after in the container; used during deletion
-	MinTxRate     *int   `json:"min_tx_rate"`          // Mbps, 0 = disable rate limiting
-	MaxTxRate     *int   `json:"max_tx_rate"`          // Mbps, 0 = disable rate limiting
-	SpoofChk      string `json:"spoofchk,omitempty"`   // on|off
-	Trust         string `json:"trust,omitempty"`      // on|off
-	LinkState     string `json:"link_state,omitempty"` // auto|enable|disable
+	MinTxRate     *int   `json:"min_tx_rate"`            // Mbps, 0 = disable rate limiting
+	MaxTxRate     *int   `json:"max_tx_rate"`            // Mbps, 0 = disable rate limiting
+	SpoofChk      string `json:"spoofchk,omitempty"`     // on|off
+	OrigSpoofChk  string `json:"origSpoofchk,omitempty"` // on|off, represents the original spoofchk value in the system
```

Comment:
I am getting concerned about littering all "non-user-configurable" parameters with corresponding json encoding fields so that we could save them on disk.
Ideally, we should only have fields with json tag if the value is expected from user. I was wondering if there's any better way of doing this.

Reply:
This regards a more general problem, for this PR we deal with the issue which was introduced in
https://github.com/intel/sriov-cni/commit/020e04d1c947619518e66bf1a59794f68bc396f7


ideally the VF's "restore" state should be in a separate cache entry (maybe different file/ different section in the file).
these values would still have json tags as the state will eventually end up as JSON to be consumed later by cmdDel

Reply:
I agree with @ahalim-intel that we should not expose a parameter that's not configurable by user. This may better go into the conf file saved in data dir. Shall we address the VF state restoration in one [issue](https://github.com/intel/sriov-cni/issues/115), which would solve all the relevent open issues such as https://github.com/intel/sriov-cni/pull/113, https://github.com/intel/sriov-cni/pull/116 etc

Reply:
I dont understand why this differs from NetConf.AdminMAC , NetConf.EffectiveMAC (These are not user configurable parameters) this patch takes the same approach to deal with what i consider a critical issue as Pod 2 Pod traffic will always fail on VFs from Mellanox NIC due to the fact that the original state of spoofchk is not preserved.

Now, don't get me wrong, i'm all for solving the general [issue](https://github.com/intel/sriov-cni/issues/115) but it can be regarded as an enhancement (crucial one IMO) to the project that will take some time to agree on design, implementation etc.
This patch is proposed as a triage until #115 is implemented.

Reply:
Also i'm not sure about backpoting policies for this project in case there was a release with this issue.

Reply:
I think the idea of not having `origSpoofChk` in `conf` also applies to `AdminMac` and `EffectiveMac`, it's just that it was not explicitly called out. Maybe we shall put all the VF state related options which are not expected to be configured by user into a separate structure or as a sub-structure of `conf` so that CNI can cache all the VF states and restore them when `cmdDel`  is called.

Reply:
Regarding backporting, I think we don't have much request as such before so there is no established rules of how this should be done. but we're open to backport any fix to release branch where needed.